### PR TITLE
trivial-builders: add onlyBin

### PR DESCRIPTION
**Pull request by risicle**
## Description of changes

Use-case described in comment, but the particular situation that spurred this was realizing that `awscli` injects its `PYTHONPACKAGES` into a nix-shell, clearly undesirable for python developers, and because `awscli` isn't a split-output package (python packages have problems with split-bin, another subject) this can't be worked around by just using `awscli.bin`.

Even if it were readily fixable for a particular package, it's very inconvenient for an end-user to have to modify an upstream package, particularly if it's heavily depended-upon and causes a lot of rebuilds.

With this, it's possible to do `nix-shell -p 'onlyBin awscli'` and have a clean `PYTHONPATH`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by lilyinstarlight — APPROVED** [review]
The `onlyBin` builder looks good to me implementation-wise and it helps with Python stuff! (e.g. `desktopToDarwinBundle` setup hook not working with packages that rely on a non-default Python, which I want to use this trivial builder to fix)

I think someone with more authority/knowledge over our trivial builders should review this more authoritatively, however, before merge

File: pkgs/build-support/trivial-builders/default.nix
Comment (risicle) [reply pkgs/build-support/trivial-builders/default.nix]: (let me squash)

**Comment by risicle** [comment]
@ofborg eval

**Review by pbsds — APPROVED** [review]

**Pull request merged**
